traveler-ios-ui/TravelerKitUI/ViewControllers/WishlistResultViewController.swift

```diff
@@ -0,0 +1,221 @@
+        let catalogItem = wishlistResult!.items[indexPath.row]
+
+        switch catalogItem {
+        case .none:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: loadingCollectionCellIdentifier, for: indexPath)
+            cell.contentView.subviews.forEach({ $0.startShimmering() })
+            return cell
+        case .some(let item):
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: wishlistCellIdentifier, for: indexPath) as! WishlistCell
+
+            cell.contentView.layer.cornerRadius = 2.0
+            cell.contentView.layer.masksToBounds = true
+
+            cell.layer.shadowColor = UIColor.lightGray.cgColor
+            cell.layer.shadowOffset = CGSize(width: 0, height: 1.0)
+            cell.layer.shadowRadius = 2.0
+            cell.layer.shadowOpacity = 1.0
+            cell.layer.masksToBounds = false
+            cell.layer.shadowPath = UIBezierPath(roundedRect: cell.bounds, cornerRadius: cell.contentView.layer.cornerRadius).cgPath
+
+            cell.contentView.backgroundColor = .white
+
+            cell.titleLabel.text = item.title
+            cell.descriptionLabel.text = item.subTitle
+            cell.priceLabel.text = "Price starting at: " + (item.price.localizedDescriptionInBaseCurrency ?? "")
+
+            if let url = item.imageURL {
+                AssetManager.shared.loadImage(with: url) { [weak cell] (image) in
+                    cell?.thumbnailImageView.image = image
+                }
+            }
+
+            cell.delegate = self
+
+            return cell
+        }
+    }
+
+    // MARK: UICollectionViewDelegate
+
+    override open func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        selectedCatalogItem = wishlistResult!.items[indexPath.row]
+        collectionView.deselectItem(at: indexPath, animated: true)
+        performSegue(withIdentifier: "itemSegue", sender: nil)
+    }
+}
+
+extension WishlistResultViewController: UICollectionViewDataSourcePrefetching {
+    public func collectionView(_ collectionView: UICollectionView, prefetchItemsAt indexPaths: [IndexPath]) {
+        guard let wishlistResult = wishlistResult else {
+            return
+        }
+
+        let indexesRequested = indexPaths.map({ $0.row }).filter({ wishlistResult.items[$0] == nil })
+        let pages = Set(indexesRequested.map({ $0 / WishlistQuery.pageSize }))
+        let pagesNotRequested = pages.filter({ !wishlistResult.hasPage($0, pageSize: WishlistQuery.pageSize) })
+        for page in pagesNotRequested where !pagesLoading.contains(page) {
+            pagesLoading.insert(page)
+
+            let query = WishlistQuery(page: page, from: wishlistResult.fromDate, to: wishlistResult.toDate)
+            Traveler.fetchWishlist(query, identifier: page, delegate: self)
+        }
+    }
```

Comment:
This is now the second place we are using this. I think its a good opportunity to think about how we can refactor this code. it might come up again

Reply:
TBD